Subproj split

Grillo-0 opened the pull request:
This is a big one, a lot of attention is needed to keep everything working

Grillo-0 committed:
Separate each project into its own directory

The project is growing, now we have two projects (vinumc and vunit) in
the future we're going to have more. To organize this, turn this repo
into a monorepo[1], were all vinum related projects reside.

The Autools build system is not up to the task (the SUBDIRS situation
is really painful when doing something like this).

So, change the build system to meson[2], using it's subproject
feature[3] makes the whole thing really manageable. Also, not having a
bunch of build scripts scattered around the project is a positive.

[1]: https://en.m.wikipedia.org/wiki/Monorepo 
[2]: https://mesonbuild.com/
[3]: https://mesonbuild.com/Subprojects.html

Fixes: #25
Fixes: #34
Fixes: #34

Grillo-0 committed:
README.md: Describe each subproject

Grillo-0 committed:
vunit: Create a README.md

Grillo-0 committed:
vinumc: Create README.md

artP2 commented on README.md:
```diff
@@ -2,40 +2,58 @@
 This is how you use it.
 
-One the project root, use `autoreconf` to generate the `./configure` script.
+One the project root, use this command to create the build directory.
```
```suggestion
On the project root, use this command to create the build directory.
```

Grillo-0 committed:
vunit: Fix TAP version string

The TAP version string is not case insensitive, we currently print
"Version", but it should be "version".

artP2 commented on meson.build:
```diff
@@ -0,0 +1,10 @@
+project('vinum', 'c',
```
add meson_version: '>= 1.1', to fix the warning bellow.
> vinumc| WARNING: Project specifies no minimum version but uses features which were added in versions:
vinumc| * 1.1: {'meson.options file'}

artP2 commented:
Add instructions about vin2dot and also how to run the tests.

artP2 reviewed (approved):
LGTM
